api/v1/logic.go

```diff
@@ -54,6 +54,12 @@ func renderRuntimeTransaction(storageTransaction client.RuntimeTransaction) (api
 		Body:       body,
 		Success:    cr.IsSuccess(),
 	}
+	if !cr.IsSuccess() {
+		code := int(cr.Failed.Code)
+		apiTransaction.Code = &code
+		apiTransaction.Module = &cr.Failed.Module
```

Comment:
I don't think we want to concat this with the `message`.

Reply:
Agreed; the payload won't be much bigger, and in general we aspire to stick close to oasis-core types and not munge data where it's not clearly beneficial.